.circleci/config.yml

```diff
@@ -12,8 +12,8 @@ jobs: # a collection of steps
           key: dependency-cache-{{ checksum "yarn.lock" }}
 
       - run:
-          name: yarn-install
-          command: yarn
+          name: yarn-install-if-has-cache
+          command:  test -d node_modules/truffle || yarn
```

Comment:
it doesn't check the package.json. it relies on pushing the yarn.lock.
I simply got fed up with dead slow commits. 
the full solution will require extracting dependencies, devDependencies from package.json and using them as part of the cache hash (so that other package.json changes will not invalidate the cache)

Reply:
Also 'yarn-install-if-**_no_**-cache' I guess?